sympy/utilities/tests/test_codegen.py

```diff
@@ -1506,3 +1506,28 @@ def test_custom_codegen():
     result = codegen(('expr', expr), header=False, empty=False, code_gen=gen)
     source = result[0][1]
     assert source == expected
+
+def test_c_with_printer():
+    #issue 13586
+    from sympy.printing.ccode import CCodePrinter
```

Comment:
maybe the test can subclass `CCodePrinter` and do something differently than the standard printer (so that we can test the string for this difference)

Reply:
@bjodah  should I use CXX98CodePrinter or something from cxxcode ??